## Index of Multiple Deprivation etc - data exploration



The data file has been copied and modified slightly to make this easier to work with and is called example.csv. This is just an example investigation and you are welcome to explore and use any of the data files provided. 




In [2]:
!ls

JA556091_TMA02_Question2b.ipynb		tm351_tma02_readme.ipynb
TMA02_Q2_Load_and_Explore_Data.ipynb	your_csv_file.csv
TMA02_Q2_Load_and_Visualise_Data.ipynb


In [3]:
!ls '../2020J_TMA02_data/Deprivation_Index'

File_10_ID2015_Local_Authority_District_Summaries.xlsx
File_11_ID_2015_Upper-tier_Local_Authority_Summaries.xlsx
File_2_ID_2015_Domains_of_deprivation.xlsx
File_3_ID_2015_Supplementary_Indices_-_Income_Deprivation_Affecting_Children_Index_and_Income_Deprivation_Affecting_Older_People_Index.xlsx
File_4_ID_2015_Sub-domains_of_deprivation.xlsx
Lower_Layer_Super_Output_Area__2011__to_Upper_Tier_Local_Authorities__2018__Lookup_in_England_and_Wales_v2.csv
example.csv


In [4]:
!pwd

/resources/DataScience/2020J_TMA02


In [5]:
import pandas as pd

# furl = "Alldata2.csv"
furl = '../2020J_TMA02_data/Deprivation_Index/example.csv'

df = pd.read_csv(furl)
df.head()

LSOAcode2011 LSOAname2011 LocalAuthorityDistrictcode2013  \
0    E01031349    Adur 001A                      E07000223   
1    E01031350    Adur 001B                      E07000223   
2    E01031351    Adur 001C                      E07000223   
3    E01031352    Adur 001D                      E07000223   
4    E01031370    Adur 001E                      E07000223   

  LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
0                           Adur                         21,352   
1                           Adur                          8,864   
2                           Adur                         22,143   
3                           Adur                         17,252   
4                           Adur                         15,643   

   IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile EmploymentRank  \
0                                 7     18,992             6         19,305   
1                                 3      9,233             3          7,879   
2                                 7     24,539             8         23,389   
3                                 6     16,087             5         13,699   
4                                 5     17,918             6         13,322   

   EmploymentDecile EducationSkillsandTrainingRank   \
0                 6                          13,727   
1                 3                           6,969   
2                 8                          10,213   
3                 5                          10,468   
4                 5                           7,819   

   EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
0                                 5                             25,876   
1                                 3                              6,883   
2                                 4                             24,693   
3                                 4                             19,627   
4                                 3                             14,893   

   HealthDeprivationandDisabilityDecile CrimeRank  CrimeDecile   \
0                                     8    12,817             4   
1                                     3    12,781             4   
2                                     8     9,112             3   
3                                     6    16,127             5   
4                                     5    13,723             5   

  BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
0                            28,166                                    9   
1                            11,399                                    4   
2                            24,743                                    8   
3                            19,559                                    6   
4                            22,433                                    7   

  LivingEnvironmentRank   LivingEnvironmentDecile   
0                 18,367                         6  
1                 16,242                         5  
2                 22,299                         7  
3                 22,819                         7  
4                 18,191                         6

In [6]:
# Preview the values in one of the data files
!tail '../2020J_TMA02_data/Deprivation_Index/example.csv'

E01013449,York 022F,E06000014,York,"19,988",7,"17,540",6,"20,349",7,"15,108",5,"22,263",7,"12,664",4,"12,629",4,"26,250",8
E01013365,York 023A,E06000014,York,"28,286",9,"25,062",8,"26,241",8,"29,820",10,"27,128",9,"31,032",10,"12,245",4,"18,787",6
E01013378,York 023B,E06000014,York,"29,849",10,"32,771",10,"32,834",10,"24,246",8,"30,520",10,"32,505",10,"9,046",3,"15,722",5
E01013450,York 023C,E06000014,York,"25,852",8,"28,355",9,"21,618",7,"30,545",10,"30,689",10,"31,475",10,"11,629",4,"7,327",3
E01013340,York 024A,E06000014,York,"32,752",10,"32,644",10,"31,055",10,"32,434",10,"30,915",10,"30,442",10,"31,186",10,"23,304",8
E01013341,York 024B,E06000014,York,"29,241",9,"23,882",8,"25,735",8,"28,951",9,"22,059",7,"32,556",10,"21,911",7,"24,596",8
E01013342,York 024C,E06000014,York,"25,821",8,"28,669",9,"30,349",10,"30,892",10,"28,677",9,"32,281",10,"2,098",1,"16,385",5
E01013423,York 024D,E06000014,York,"32,637",10,"32,214",10,"31,726",10,"31,757",10,"31,363",10,"30,384",10,"21,452",7,"27

This file that has been loaded has a number of columns. It is worth seeing what type of columns are available for the data



In [7]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

In [8]:
ListofNames = df['LocalAuthorityDistrictname2013'].unique()
ListofNames

array(['Adur', 'Allerdale', 'Amber Valley', 'Arun', 'Ashfield', 'Ashford',
       'Aylesbury Vale', 'Babergh', 'Barking and Dagenham', 'Barnet',
       'Barnsley', 'Barrow-in-Furness', 'Basildon',
       'Basingstoke and Deane', 'Bassetlaw',
       'Bath and North East Somerset', 'Bedford', 'Bexley', 'Birmingham',
       'Blaby', 'Blackburn with Darwen', 'Blackpool', 'Bolsover',
       'Bolton', 'Boston', 'Bournemouth', 'Bracknell Forest', 'Bradford',
       'Braintree', 'Breckland', 'Brent', 'Brentwood',
       'Brighton and Hove', 'Bristol, City of', 'Broadland', 'Bromley',
       'Bromsgrove', 'Broxbourne', 'Broxtowe', 'Burnley', 'Bury',
       'Calderdale', 'Cambridge', 'Camden', 'Cannock Chase', 'Canterbury',
       'Carlisle', 'Castle Point', 'Central Bedfordshire', 'Charnwood',
       'Chelmsford', 'Cheltenham', 'Cherwell', 'Cheshire East',
       'Cheshire West and Chester', 'Chesterfield', 'Chichester',
       'Chiltern', 'Chorley', 'Christchurch', 'City of London',
       'Co

the list can be used to find the areas that you are interested in 

In [9]:
df[df.LocalAuthorityDistrictname2013 == 'Bradford'].head()

LSOAcode2011   LSOAname2011 LocalAuthorityDistrictcode2013  \
3095    E01010646  Bradford 001A                      E08000032   
3096    E01010647  Bradford 001B                      E08000032   
3097    E01010648  Bradford 001C                      E08000032   
3098    E01010692  Bradford 001D                      E08000032   
3099    E01010691  Bradford 002A                      E08000032   

     LocalAuthorityDistrictname2013 IndexofMultipleDeprivationRank  \
3095                       Bradford                         30,426   
3096                       Bradford                         26,378   
3097                       Bradford                         29,500   
3098                       Bradford                         29,485   
3099                       Bradford                         32,725   

      IndexofMultipleDeprivationDecile IncomeRank  IncomeDecile  \
3095                                10     27,564             9   
3096                                 9     20,319             7   
3097                                 9     27,109             9   
3098                                 9     30,820            10   
3099                                10     31,852            10   

     EmploymentRank  EmploymentDecile EducationSkillsandTrainingRank   \
3095         29,855                10                          31,470   
3096         22,965                 7                          28,414   
3097         27,437                 9                          30,579   
3098         26,796                 9                          32,677   
3099         32,530                10                          32,786   

      EducationSkillsandTrainingDecile HealthDeprivationandDisabilityRank  \
3095                                10                             22,927   
3096                                 9                             12,912   
3097                                10                             19,905   
3098                                10                             22,549   
3099                                10                             27,915   

      HealthDeprivationandDisabilityDecile CrimeRank  CrimeDecile   \
3095                                     7    32,447            10   
3096                                     4    31,074            10   
3097                                     7    32,574            10   
3098                                     7    23,108             8   
3099                                     9    29,647            10   

     BarrierstoHousingandServicesRank   BarrierstoHousingandServicesDecile   \
3095                            25,639                                    8   
3096                            31,616                                   10   
3097                            19,208                                    6   
3098                            20,126                                    7   
3099                            29,258                                    9   

     LivingEnvironmentRank   LivingEnvironmentDecile   
3095                 14,518                         5  
3096                 25,406                         8  
3097                 21,196                         7  
3098                 16,628                         6  
3099                 25,899                         8

The local authority district code might be interesting later or the area code.

In [ ]:
df['LocalAuthorityDistrictcode2013'].unique()

## Once you have an area of interest you might want to filter on that area. Here is an example using Bradford

In [ ]:
df.query('LocalAuthorityDistrictname2013 =="Bradford"')

There are over 57 columns - it would be good to have a list of what type of data is being held - remember list(df.columns)

In [ ]:
list(df.columns)

The reports provided explain about how the numerical values - how they are calculated and their significance. 
The lower the value e.g.  the number 1 is the most deprived in the Index of Multiple Deprivation factors.

This example looks at extracting for Bradford the those areas with a value of 1


In [ ]:
df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile == 1')

Let's look a bit more the data in general - linked to the Index of Multiple Deprivation is the rank value. 

In [10]:
df_wide = df.pivot(index= 'IndexofMultipleDeprivationDecile', columns='LSOAname2011', values='IndexofMultipleDeprivationRank')
df_wide.head()

LSOAname2011                     Adur 001A Adur 001B Adur 001C Adur 001D  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN     8,864       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                      NaN       NaN       NaN       NaN   

LSOAname2011                     Adur 001E Adur 001F Adur 002A Adur 002B  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN       NaN       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                   15,643       NaN       NaN       NaN   

LSOAname2011                     Adur 002C Adur 002D  ... York 022F York 023A  \
IndexofMultipleDeprivationDecile                      ...                       
1                                      NaN       NaN  ...       NaN       NaN   
2                                      NaN       NaN  ...       NaN       NaN   
3                                      NaN       NaN  ...       NaN       NaN   
4                                      NaN       NaN  ...       NaN       NaN   
5                                      NaN       NaN  ...       NaN       NaN   

LSOAname2011                     York 023B York 023C York 024A York 024B  \
IndexofMultipleDeprivationDecile                                           
1                                      NaN       NaN       NaN       NaN   
2                                      NaN       NaN       NaN       NaN   
3                                      NaN       NaN       NaN       NaN   
4                                      NaN       NaN       NaN       NaN   
5                                      NaN       NaN       NaN       NaN   

LSOAname2011                     York 024C York 024D York 024E York 024F  
IndexofMultipleDeprivationDecile                                          
1                                      NaN       NaN       NaN       NaN  
2                                      NaN       NaN       NaN       NaN  
3                                      NaN       NaN       NaN       NaN  
4                                      NaN       NaN       NaN       NaN  
5                                      NaN       NaN       NaN       NaN  

[5 rows x 32844 columns]

A quick glance (although a better filter and table could be used here) - shows that areas e.g. Adur and York rank value above 4000 puts in the deprivation value above 2 - this is just glancing at the data. Other filters and queries can be developed to check this. Data could be organised to investigate to see for example which factors might at the most deprived level are impacting the chosen area.

If we review the document The English Index of Multiple Deprivation
(IMD) 2015 – Guidance

see page 2 "Small area X is ranked 5,000 out of 32,844 small-areas in England, where 1 is the most deprived. This means that small area X is amongst the 20 per cent most deprived small areas in the country."

See page 10 of The English Indices of Deprivation 2015 – Frequently Asked Questions (FAQs) to see the table of ranking

Quick look at the data to see if anything interesting 

In [ ]:
df_wide.describe()

Let's focus on Bradford filter for IMD of 1 and 2

In [ ]:
tmp = df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile < 3')
tmp1_wide = tmp.pivot(index= 'IndexofMultipleDeprivationDecile', columns='LSOAname2011', values='IndexofMultipleDeprivationRank')
tmp1_wide.head()

From initial analysis of the IMD of there are 131 areas in Bradford that fall into the IMD of 1 and 2. We have over a number of columns that would be interesting to investigate.

In [11]:
list(df.columns)

['LSOAcode2011',
 'LSOAname2011',
 'LocalAuthorityDistrictcode2013',
 'LocalAuthorityDistrictname2013',
 'IndexofMultipleDeprivationRank',
 'IndexofMultipleDeprivationDecile',
 'IncomeRank',
 'IncomeDecile',
 'EmploymentRank',
 'EmploymentDecile',
 'EducationSkillsandTrainingRank ',
 'EducationSkillsandTrainingDecile',
 'HealthDeprivationandDisabilityRank',
 'HealthDeprivationandDisabilityDecile',
 'CrimeRank',
 'CrimeDecile ',
 'BarrierstoHousingandServicesRank ',
 'BarrierstoHousingandServicesDecile ',
 'LivingEnvironmentRank ',
 'LivingEnvironmentDecile ']

You might want to explore specific data areas for an area. 
Filter the data further. Some examples are provided - this you have already seen - slightly different IMD parameter

In [ ]:
df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile > 3')

let's look at the overall data for Bradford

In [ ]:
tmp2 = df.query('LocalAuthorityDistrictname2013 =="Bradford"')
tmp3_wide = tmp2.pivot(index= 'LSOAname2011', columns='IndexofMultipleDeprivationDecile', values='IndexofMultipleDeprivationRank')
tmp3_wide.head(300)

In [ ]:
tmp3_wide.describe

Filter on more specific data e.g. Education Skills

In [ ]:
tmp3_wide = tmp2.pivot(index= 'LSOAname2011', columns= 'EducationSkillsandTrainingDecile', values='EducationSkillsandTrainingRank ')
tmp3_wide.head(50)

Data can be filtered, investigated and sumerised on the different columns. After exploring this and areas of interest then the students can start to decide on what aspects to investigate and compare.


Filter further on skills analysis for those Adults in the most deprived areas of Bradford. e.g. IMD = 1

In [ ]:
tmp2 = df.query('LocalAuthorityDistrictname2013 =="Bradford" & IndexofMultipleDeprivationDecile == 1')
tmp3_wide = tmp2.pivot(index= 'LSOAname2011',columns = 'EducationSkillsandTrainingDecile', values='EducationSkillsandTrainingRank ')
tmp3_wide.head(101)